llvm/lib/Transforms/Scalar/LICM.cpp

```diff
@@ -2247,7 +2253,7 @@ bool llvm::promoteLoopAccessesToScalars(
       PreheaderLoad->setOrdering(AtomicOrdering::Unordered);
     PreheaderLoad->setAlignment(Alignment);
     PreheaderLoad->setDebugLoc(DebugLoc());
-    if (AATags)
+    if (AATags && LoadIsGuanteedToExecute)
```

Comment:
I think we also need to check StoreIsGuaranteedToExecute for the LoopPromoter use above. Should only copy AATags if it's a non-speculative store.

Reply:
Updated, thanks.